### PR TITLE
Start an FAQ for builder authors

### PR DESCRIPTION
**Pull request by natebosch**
Towards #1431

We can add answers to this doc as they come up in issues. Start with an
answer for https://github.com/google/pageloader/issues/132

### CODE REVIEW

**Comment by natebosch** [comment]
> Should we just make this a general `faq.md`?

I think builder authors and end users are different enough audiences to have 2 docs.

File: docs/builder_author_faq.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## When should a Builder build to `cache` vs `source`?
+
+Use `build_to: source` when:
+- The generated code _does not_ depend on details outside of the package with
```
Comment (grouma) [inline docs/builder_author_faq.md]: I think I understand this first sentence but it is very hard for me to parse. Can you reword it? The multiple references to generated code specifically cause the confusion. Maybe use "input sources" for the second reference?

**Comment by natebosch** [comment]
> please add a link prominently on the main README.md for the repo

Done. PTAL

File: docs/builder_author_faq.md (hunk 1 above)
Comment (natebosch) [reply docs/builder_author_faq.md]: Rephrased. Is it more clean now?

**Pull request merged**